packages/venia-ui/lib/RootComponents/Search/search.js

```diff
@@ -1,181 +1,156 @@
+import { getFilterParams } from '@magento/peregrine/lib/util/getFilterParamsFromUrl';
 
+import { mergeClasses } from '../../classify';
 import Gallery from '../../components/Gallery';
-import classify from '../../classify';
-import Icon from '../../components/Icon';
-import { getFilterParams } from '@magento/peregrine/lib/util/getFilterParamsFromUrl';
-import getQueryParameterValue from '../../util/getQueryParameterValue';
-import isObjectEmpty from '../../util/isObjectEmpty';
-import { X as CloseIcon } from 'react-feather';
 import FilterModal from '../../components/FilterModal';
 import { fullPageLoadingIndicator } from '../../components/LoadingIndicator';
-import defaultClasses from './search.css';
 import PRODUCT_SEARCH from '../../queries/productSearch.graphql';
+import getQueryParameterValue from '../../util/getQueryParameterValue';
+import isObjectEmpty from '../../util/isObjectEmpty';
+import CategoryFilters from './categoryFilters';
+import defaultClasses from './search.css';
 
-const getCategoryName = gql`
-    query getCategoryName($id: Int!) {
-        category(id: $id) {
-            name
-        }
-    }
-`;
-
-export class Search extends Component {
-    static propTypes = {
-        classes: shape({
-            noResult: string,
-            root: string,
-            totalPages: string
-        }),
-        openDrawer: func.isRequired,
-        executeSearch: func.isRequired,
-        history: object,
-        location: object.isRequired,
-        match: object,
-        searchOpen: bool,
-        toggleSearch: func
-    };
-
-    componentDidMount() {
-        // Ensure that search is open when the user lands on the search page.
-        const { location, searchOpen, toggleSearch, filterClear } = this.props;
-
-        const inputText = getQueryParameterValue({
-            location,
-            queryParameter: 'query'
-        });
-
-        isObjectEmpty(getFilterParams()) && filterClear();
-
-        if (toggleSearch && !searchOpen && inputText) {
-            toggleSearch();
-        }
-    }
-
-    componentDidUpdate(prevProps) {
-        const queryPrev = getQueryParameterValue({
-            location: prevProps.location,
-            queryParameter: 'query'
-        });
-
-        const queryCurrent = getQueryParameterValue({
-            location: this.props.location,
-            queryParameter: 'query'
-        });
-        if (queryPrev !== queryCurrent) {
-            this.props.filterClear();
-        }
-    }
+const Search = props => {
+    const { history, location } = props;
+    const shouldClear = useRef(false);
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    // retrieve app state and action creators
+    const [appState, appApi] = useAppContext();
+    const { searchOpen } = appState;
+    const { executeSearch, toggleDrawer, toggleSearch } = appApi;
+    const [, catalogApi] = useCatalogContext();
+    const { clear: clearFilters } = catalogApi.actions.filterOption;
```

Comment:
Do you want to use the async action `filterClear` instead of this? I know you fixed the `actions` export already. Just LMK, I don't feel strongly either way.

Reply:
Decided to stick with actions like this.